org-code-javabuilder/playground/src/main/java/org/code/playground/Item.java

```diff
@@ -1,31 +1,51 @@
 package org.code.playground;
 
-interface Item {
+public abstract class Item {
```

Comment:
I think we'll want a constructor for `Item` that takes in x, y, and height

Reply:
For my understanding, can you elaborate? We would want students to be able to create ImageItems with no filename, or TextItems with no text?

Reply:
Sure thing - so because `Item` is declared abstract, it cannot be instantiated anyway. Students would still only ever instantiate `ImageItem/TextItem/ClickableImage` instead of `Item`s directly, but a constructor would allow `Item` to set some common properties that subclasses could reference. Instead of each subclass needing to set x, y, and height themselves, these would be set by the superclass which just saves some boilerplate. For example, `Item` would be implemented something like:
```
protected Item(int x, int y, int height) {
   this.x = x;
   this.y = y;
   this.height = height;
}

...

public getX() {
   return this.x;
}

public setX(int x) {
   this.x = x;
}

...
```

And then subclasses (for example, ImageItem) would be implemented like
```
public ImageItem(int x, int y, int width, int height, String imageUrl) {
   super(x, y, height);
   this.width = width;
   this.imageUrl = imageUrl;
}

...

public getX() {
   return super.getX();
}

public setX(int x) {
   super.setX(x);
}

...

public getWidth() {
   return this.width;
}
```

So ImageItem would only need to worry about explicitly getting and setting properties specific to it (in this case width and imageUrl) and defer to its parent class for everything else.

FWIW, this will come in handy later when we have these items send Playground updates; for each `set` method, we will need to have the class send an update message, and having `Item` be a base class allows `Item` to own the updating logic and subclasses can just call a superclass method, instead each of them needing to having updating logic as well.

Reply:
Great explanation, got it thank you!